### DISCUSSION TITLE
About the SPDX lisence identifier and version

### DISCUSSION THREAD

**Discussion by Noveleader** [smartcontractkit/full-blockchain-solidity-course-py#998] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/91677627/153154151-cc8fbda7-10c5-46dd-b515-08694991e6b6.png)
I have been writing this and I am not able to get the license identifier concept, and while it is in VM the Identifier doesn't matter? and which identifier I should use in real network?

And also ^0.6.0 this means upto this version or from this version?

**Comment by n4n0b1t3** [comment, marked as answer]
Hi, please always use the License Identifier. [Software Package Data Exchange (SPDX)](https://en.wikipedia.org/wiki/Software_Package_Data_Exchange) is an open standard. In a project where you want to encourage participation, the MIT license would be appropriate because it is the most open license, so everyone can just copy code and reuse it elsewhere. The value of software is often in the concept of the whole product and seldom on the code level. This is good! When I think about it, not much of my code actually comes from me thinking about it long and hard. I am searching for a working solution and adapting it to my project.

**Reply by Noveleader** [reply]
Ok got it so that means we require license to run code in solidity and we need to mention which license we are using, I think I got the point.
Thanks and also see my second question about pragma solidity <version>.

**Reply by cromewar** [reply]
Yes using a license always is very important, as the contracts get verified on block explorers and all the source code goes public is important to know how that code could be used by others.

**Reply by Noveleader** [reply]
Cool got it

**Comment by n4n0b1t3** [comment]
Pragma solidity: 
`^0.6.0` means everything within the 0.6.* range
`>=0.6.0` means everything upwards from 0.6.0 including 0.7.* and upwards

**Reply by Noveleader** [reply]
thanks got it
